store_app/views/transaction.py

```diff
@@ -0,0 +1,19 @@
+from flask import make_response, jsonify, request, g
+from app import app
+from business import transaction
+
+
+@app.route('/movies/transaction', methods=['PUT'])
+def movie_bought_update_quantity():
+    try:
+        imdb_id = request.json['imdb_id']
+        quantity_purchased = request.json['quantity']
+        if g.user.role != 'customer':
+            raise ValueError('User must be a customer to perform this function.')
+        if not g.user.id:
+            raise ValueError('User must be logged in to perform this function.')
```

Comment:
Can you be logged in and not have an id? If so there is something weird about the flow.

Reply:
I have tried to correct it. I just wanted to check if the user is in the session. So, now I just use:   " if not g.user: " instead if g.user.id